internal/pkg/instrumentation/probe/probe.go

```diff
@@ -36,17 +33,36 @@ type Probe interface {
 	// the information about the package the Probe instruments.
 	Manifest() Manifest
 
-	// Load loads all the eBPF programs and maps required by the Probe.
-	// It also attaches the eBPF programs to the target process.
-	// TODO: currently passing Sampler as an initial configuration - this will be
-	// updated to a more generic configuration in the future.
-	Load(*link.Executable, *process.Info, *sampling.Config) error
+	// Init initializes the Probe, setting up the reader, closers, and sampling config.
+	Init(*sampling.Config) error
 
 	// Run runs the events processing loop.
 	Run(func(ptrace.ScopeSpans))
 
 	// Close stops the Probe.
 	Close() error
+
+	// GetLogger returns the *slog.Logger associated with the Probe.
+	GetLogger() *slog.Logger
```

Comment:
Why would we ever need to get the logger from a probe? Doesn't the Manager have its own logger it can use for its own operations?

Reply:
Yes. Any logger that the manager passes should just be the manager's logger.

Reply:
sgtm, I wasn't totally sure what this relationship was between them